src/sources/datadog_agent/mod.rs

```diff
@@ -105,6 +105,11 @@ pub struct DatadogAgentConfig {
     #[configurable(derived)]
     #[serde(default, deserialize_with = "bool_or_struct")]
     acknowledgements: SourceAcknowledgementsConfig,
+
+    /// If this setting is set to `true`, then the source will do some additional parsing to
+    /// simplify received tags on log messages from the Datadog Agent into a common format.
+    #[serde(default = "crate::serde::default_false")]
+    simplify_log_tags: bool,
```

Comment:
Iffy on the name, knowing how it works from my end it's debatable if it's simplified... maybe something like `tags_as_object`